### ISSUE TITLE
[FEATURE] Need of a timeout option ?

### ISSUE THREAD

**Issue by davidaparicio** [ycd/dstp#22]
Hello!

I tried with a website down like [jetable.org](https://jetable.org/)
But no answer/summary from the tool dstp 💥 
Is there a timeout option ?
```
➜ dstp jetable.org
^C
~ took 8m 32s
➜ dig +short jetable.org
193.200.42.56
➜ ping jetable.org
PING jetable.org (193.200.42.56): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
^C
--- jetable.org ping statistics ---
4 packets transmitted, 0 packets received, 100.0% packet loss
```

**Comment by ycd** [comment]
Can you try this one again with sudo?

Yes, there is a PR for adding timeout support (#19) which i will review, merge and release soon!

**Comment by davidaparicio** [comment]
It's curious because I'm on Mac OS, and without sudo works ;)
```
➜ dstp 8.8.8.8
Ping: 7.030666ms
DNS: resolving 8.8.8.8
SystemDNS: resolving 8.8.8.8
TLS: certificate is valid for 55 more days
HTTPS: got 200 OK

~ took 5s
```

With sudo
```
➜ sudo dstp 8.8.8.8
Password:
Ping: 7.334667ms
DNS: resolving 8.8.8.8
SystemDNS: resolving 8.8.8.8
TLS: certificate is valid for 55 more days
HTTPS: got 200 OK


~ took 7s
➜ sudo dstp jetable.org
^C

~ took 1m 11s
```
The same, no result :(

**Comment by ycd** [comment]
Hey @davidaparicio, `jetable.org` hangs for me too and seems related to the site, even sending simple ICMP packages with `ping` fails too.  

Also just implemented timeout option and added fault tolerance for `dstp` to work in non-privileged scenarios, i'll release the new version soon.

**Comment by ycd** [comment]
Timeout feature is available with v0.3.1 🎉 🚀

**Comment by davidaparicio** [comment]
It works with and without sudo rights ;)
Thanks a lot for this feature !! 🥳 

```
➜ sudo dstp jetable.org -t 2s -p 3
Ping: no response
DNS: resolving 193.200.42.56
SystemDNS: resolving 2001:67c:1740:9000::30:56, 193.200.42.56
TLS: dial tcp [2001:67c:1740:9000::30:56]:443: connect: network is unreachable
HTTPS: Get "https://jetable.org": dial tcp [2001:67c:1740:9000::30:56]:443: connect: network is unreachable

~ took 1m 57s
➜ dstp jetable.org -t 2s -p 3
Ping: no response
DNS: resolving 193.200.42.56
SystemDNS: resolving 2001:67c:1740:9000::30:56, 193.200.42.56
TLS: dial tcp [2001:67c:1740:9000::30:56]:443: connect: network is unreachable
HTTPS: Get "https://jetable.org": dial tcp [2001:67c:1740:9000::30:56]:443: i/o timeout

~ took 1m 58s
```
